TRANSIP: Fix staticcheck errors

CC @blackshadev

I'm moving from go-lint to staticcheck (https://staticcheck.io/). It finds some issues that I think are worth fixing.

Docs on how to install staticcheck are here: https://staticcheck.io/docs/getting-started/
The command line that produced these errors was: `staticcheck ./...`

The warnings/errors related to this provider are:

```
providers/transip/transipProvider.go:210:20: *&x will be simplified to x. It will not copy x. (SA4001)
```

--- a/providers/transip/transipProvider.go
+++ b/providers/transip/transipProvider.go
@@ -207,7 +207,7 @@ func recordToNative(config *models.RecordConfig) (domain.DNSEntry, error) {
 
 func nativeToRecord(entry domain.DNSEntry, origin string) (*models.RecordConfig, error) {
 	rc := &models.RecordConfig{
-		TTL:      uint32(*&entry.Expire),
+		TTL:      uint32(entry.Expire),
 		Type:     entry.Type,
 		Original: entry,
 	}
@@ -217,16 +217,6 @@ func nativeToRecord(entry domain.DNSEntry, origin string) (*models.RecordConfig,
 	}
 
 	return rc, nil
-}
-
-func removeNS(records models.Records) models.Records {
-	var noNameServers models.Records
-	for _, r := range records {
-		if r.Type != "NS" {
-			noNameServers = append(noNameServers, r)
-		}
-	}
-	return noNameServers
 }
 
 func removeOtherNS(dc *models.DomainConfig) {

